BHSRobotix/ChargedUp2023
# Granite state

Lot's of tweaks/changes resulting from the Granite State Competition:

- PathPlanner
   - Fixed PathPlanner automatic path translation based on current alliance.  This allows us to generate just paths on the blue side, and it will automatically (and correctly) be translated for the red side
   - Updated PathPlanner to latest version
- Simulation Improvements
   - Wrapped camera settings in "hasCamera" capability. Fixed simulation crash.
   - Only bind joystick commands if joysticks exist. This fixes the flood of console messages when the joysticks aren't connected (e.g. in simulation mode).
- LED
   - Set LED color to match current alliance
   - Fixed CONE/CUBE enum and set LED color to match if cone/cube is selected for pickup
- Autonomous
   - Do not stop robot at the end of robot driving commands. This allows us to smoothly sequence multiple commands together
   - Explicitly stop robot after the end a sequence of autonomous commands
   - Run DriveStraightToDock between PathPlanner and BalancePID to ensure we are on ramp.
   - In DriveStraightPID and RotateDegrees commands, set the rotationPid to use continuous input (0 to 360).  _This fixes the spinning in circles issue_. 
   - Keep track of starting distance in DriveStraightPID for use in distancePid. _This fixes using DriveStraightPID in a sequence_
- Misc
   - Set visual camera resolution to improve responsiveness. 
 
Still Todo:
- [x] Handle exception when mac address returns null (when not connected to network).
- [x] Print error if joysticks aren't connected on **competition** bot
